### ISSUE TITLE
On/Of state issue

### ISSUE THREAD

**Issue by oleyek** [seikan/homebridge-sonoff-basic-espeasy#6]
The status On/Off in Homebridge is not changing when I switch on/off pressing physical button or change state through http request.

**Comment by pbksol** [comment]
The state is only updated when you open the Home-App. It's not possible to receive state changes from a device without a platform plugin containing a server to push changes to. Correct me if I'm wrong.

**Comment by oleyek** [comment]
Yes, you're right. Maybe the better way to do this will be using MQTT like in plugin for homebridge for Tasmota firmware?

**Comment by erokoder** [comment]
I'm pretty sure it's impossible to changing when pressing physical button. If you know way, share with us.

**Comment by oleyek** [comment]
It's possible with MQTT, because when state is changed switch sending notification to MQTT broker.

**Comment by erokoder** [comment]
Could you explain what way sonoff with MQTT know that wall switch change status?

**Comment by pbksol** [comment]
It's nothing related to this project. But… You need an additional device, a "MQTT broker" and you have to configure it to communicate with each MQTT device. You can do a search on Google on how to configure this things. Spoiler: It's a bag of hurts.
The way it works? The same as with Homebridge Platforms. MQTT devices report their state to their broker. This state is then pushed to Homebridge and HomeKit.
You can configure MQTT with ESPeasy, there's no Tasmota firmware necessary. But that's a completely different story…

**Comment by erokoder** [comment]
I'm not sure you understand me.
I have this scheme wiring: http://prntscr.com/i0qrfb

Test case:
State OFF on HomeAPP.
1. Turn on Lights with my HomeAPP  on R1 relay, and Sonoff PRO swithc reley and Light ON.
State ON on HomeAPP.
2. When I pressing physical key "WALL SWITHC" Disconnect Electric Cricuit and turn off Light.
State continue ON on HomeAPP. 

How to change state, which device to purchase?

**Comment by pbksol** [comment]
I do understand. You have two own Homebridge Repos so I assume that you do know how the basics of homebridge work and what the differences between accessories and platforms are. That difference is the reason why the state does not change when you press the physical button.
If you want to change that behaviour you need either a homebridge platform or some server infrastructure for the Sonoff to report (push) state changes to which then needs to be integrated into HomeKit/-Bridge. One option is MQTT or something similar – how to configure this is something to ask the MQTT-people, ESPeasy offers support for it by default.
But there's no way to change the behaviour of this plugin (as long as it's an accessory) to detect the changes. The state of an accessory is only updated when HomeKit polls for it's state and there's no way for an accessory to initiate the update by itself.
